### ISSUE TITLE
Establish a prioritization model

### ISSUE THREAD

**Issue by timfoote** [daveshap/raven#38]
As a project team, we need to determine a method for prioritizing our work items in order to effectively allocate resources and meet project goals. The prioritization technique we select should provide clear guidance for determining which work items are most important, and help us make informed decisions about which work items to tackle first. The chosen technique should also be flexible enough to accommodate changes in project requirements and be easily understood by all members of the team.

**Comment by timfoote** [comment]
I've previously been a fan of MoSCoW

![image](https://user-images.githubusercontent.com/878550/216803564-98c7cef0-d3b9-4164-981d-93c4cf340784.png)

**Comment by daveshap** [comment]
This is great stuff. I haven't yet formed a preference, but here's how I think of it:

- Prioritization is in layers, with some things (like the nexus) at the very core of decision-making and functionality
- The innermost layer is a necessary prerequisite technology/decision
- The next layer is everything that is immediately dependent upon the core layer (blocked)
- So on and so forth up to the outermost layers

As RAVEN/MARAGI is a "cognition first" model, the output layer is actually one of the least important (still important) but it comes last due to the fact that all the cognition, planning, and decision making has to happen first. Another way to think of this is more like the OSI model with layers of abstraction.

Maybe this is something I need to do to help communicate the architecture. This won't dictate prioritization but it will help frame stuff... Let me take a crack at it.

**Comment by timfoote** [comment]
The architecture is a hot topic I'm keen to discuss more, as I can help put together different viewpoints for different audiences, and express the architecture through various frameworks (archimate, TOGAF, etc) and patterns. Sounds like some decision trees or matrixes could be helpful also. 

To be continued...

**Comment by twcook** [comment]
@timfoote, excellent work.

**Comment by BASIC-JJ** [comment]
Love all of the prioritization techniques here!  Not much to add here besides the fact I’m a big fan of MSCW and feel it’s a good all-rounder for product development.

Also, it may be beneficial to use Prioritization Quadrant in conjunction with MSCW for the initial development of RAVEN R1/R2.

**Comment by Ben-Epstein** [comment]
I really like the idea of combining mscw and the Complexity X Value matrix. This would allow us to both understand what we _need_ to build now, and figure out who should work on what. 

For example, 2 equally important items may be must haves, but one may be significantly lower complexity, and could be given to someone with less time, or less experience. It aligns well with the `good_firsr_issue` tag, which is a really valuable tag to have in open source projects. 

If we didn't want the overhead of both, we could also just stick with MSCW, but enforce a tagging system of every item, 1-5, defining complexity.

**Comment by priuatus** [comment]
Hey everyone! I had no idea these techniques and methods for prioritization of issues even existed. Defining a system makes sense knowing how OSS issues tend to balloon out of control as projects get traction . 

My question is: Assuming the chosen technique or method can be well specified and deterministic. Can we prompt or tune a LLM like GPT to classify and label issues based on the chosen method? If so then github workflows can be used to automate initial issue labeling.

If it's possible and out of scope for this project. I assume there are a lot of projects that would find value in automating some of the chores around issue labeling and prioritization. Would it be a good standalone AI project?
